adapters/smartyads/smartyads.go

```diff
@@ -0,0 +1,209 @@
+package smartyads
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+// Implements Bidder interface.
+type SmartyAdsAdapter struct {
+	endpoint template.Template
+}
+
+func NewSmartyAdsBidder(endpointTemplate string) *SmartyAdsAdapter {
+	template, err := template.New("endpointTemplate").Parse(endpointTemplate)
+	if err != nil {
+		return nil
+	}
+	return &SmartyAdsAdapter{endpoint: *template}
+}
+
+func (a *SmartyAdsAdapter) CheckHasImps(request *openrtb.BidRequest) error {
+	if len(request.Imp) == 0 {
+		err := &errortypes.BadInput{
+			Message: "Missing Imp Object",
+		}
+		return err
+	}
+	return nil
+}
+
+func GetHeaders(request *openrtb.BidRequest) *http.Header {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("X-Openrtb-Version", "2.5")
+
+	if request.Device != nil {
+		if len(request.Device.UA) > 0 {
+			headers.Add("User-Agent", request.Device.UA)
+		}
+
+		if len(request.Device.IP) > 0 {
```

Comment:
You may wish to also consider that `request.Device.IPv6` may be set when this field is empty.